### PR TITLE
feat: Add error handling

### PR DESCRIPTION
**Pull request by deepakvenkat**
**What**
---
Adds error handling and universal not found and unknown error pages

**Why**
----
* In order to ensure the app does not show a stack trace to the user but instead shows a proper 404 and 500 pages

**Clubhouse**
---
https://app.clubhouse.io/out-in-tech/story/337/fe-400-errors-page
https://app.clubhouse.io/out-in-tech/story/338/fe-500-error-page

**How**
---
* Add `NotFound` and `UnknownError` components
* Add an `ErrorBoundary` component
* Wrap all route content in ErrorBoundary
* Add a catch all `useError` hook. This is needed to throw and catch errors from `useEffect` hooks
* Add a `wrappedFetch` util. This is because fetch does not throw when there is a status error. This wrappedFetch will throw for anything > 299
* Update all api to use the wrappedFetch util.
* Update space details to use throwError

**How was it tested**
---
* Locally by going to a page which does not exist and a space which does not exist
Screen Shot 2021-03-21 at 7 35 12 PM
Screen Shot 2021-03-21 at 7 35 34 PM

* Locally making the backend throw a 500
Screen Shot 2021-03-21 at 7 36 09 PM

### CODE REVIEW

File: src/api/wrappedFetch.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import fetch from 'node-fetch';
+
+const wrappedFetch = async (url, opts) => {
```
Comment (rachel-fischoff) [inline src/api/wrappedFetch.js:3]: This is great! Thanks for adding it to all the api files!

File: src/hooks/useError.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { useState, useCallback } from 'react';
+
+const useAsyncError = () => {
+  // eslint-disable-next-line no-unused-vars
+  const [_, setError] = useState();
```
Comment (rachel-fischoff) [inline src/hooks/useError.js:5]: Not sure what the _ means?

**Review by rachel-fischoff — APPROVED** [review]
Looks good! I just had question in the useError file! Left the comment on the line!

File: src/hooks/useError.js (hunk 2 above)
Comment (deepakvenkat) [reply src/hooks/useError.js:5]: That's just a convention when you don't care about the first parameter. javascript does not allow you to do `const [, setError] = useState();` so `_` usually signifies an unused param.

**Pull request merged**